src/debugSession/BrightScriptDebugSession.ts

```diff
@@ -670,21 +670,135 @@ export class BrightScriptDebugSession extends BaseDebugSession {
+                        }
+                    }
+                }
+                this.sendEvent(new OutputEvent(line, 'stdout'));
+                this.sendEvent(new LogOutputEvent(line));
+            }
+        });
+        return this.pendingSendLogPromise;
+    }
+
+    // https://regex101.com/r/ixpQiq/1
+    private deviceFilePathRegex = /((?:\.\.\.|[A-Za-z_0-9]*pkg\:\/)[A-Za-z_0-9 \/\.]+\.[A-Za-z_0-9 \/]+)(?:(?:\:)(\d+)(?:\:(\d+))?|\((\d+)(?:\:(\d+))?\))/ig;
+
+    /**
+     * Extracts potential package paths from a given line of text.
+     *
+     * This method uses a regular expression to find matches in the provided line
+     * and returns an array of objects containing details about each match.
+     *
+     * @param input - The line of text to search for potential package paths.
+     * @returns An array of objects, each containing:
+     *   - `fullMatch`: The full matched string.
+     *   - `path`: The extracted path from the match.
+     *   - `lineNumber`: The line number extracted from the match.
+     *   - `columnNumber`: The column number extracted from the match, or `undefined` if not found.
+     */
+    private getPotentialPkgPaths(input: string): Array<{ fullMatch: string; path: string; lineNumber: number; columnNumber: number }> {
+        let matches = input.matchAll(this.deviceFilePathRegex);
+        let paths: ReturnType<BrightScriptDebugSession['getPotentialPkgPaths']> = [];
+        if (matches) {
+            for (let match of matches) {
+                let fullMatch = match[0];
+                let path = match[1];
+                let lineNumber = parseInt(match[2] ?? match[4]);
+                let columnNumber = parseInt(match[3] ?? match[5]);
+                if (isNaN(columnNumber)) {
+                    columnNumber = undefined;
+                }
+                paths.push({
+                    fullMatch: fullMatch,
+                    path: path,
+                    lineNumber: lineNumber,
+                    columnNumber: columnNumber
+                });
             }
-            this.sendEvent(new OutputEvent(line, 'stdout'));
-            this.sendEvent(new LogOutputEvent(line));
         }
+        return paths;
+    }
+
+    // https://regex101.com/r/Gl87iT/1
+    private deviceBacktraceObjectRegex = /{\s+filename:\s+"([A-Za-z0-9_\.\/\:]+)"\s+function\:\s+".+"\s+line_number\:\s+(\d+)\s+}/gi;
+
+    /**
+     * Converts the filename property in backtrace objects in the given input string to source paths if found
+     */
+    private async convertBacktracePaths(input: string) {
+        if (!this.launchConfiguration.enableConsolePkgToSourcePathConversion) {
+            return input;
+        }
+        // Why does this not work? It should work, but it doesn't. I'm not sure why.
+        // let matches = input.matchAll(this.deviceBacktraceObjectRegex);
+
+        this.deviceBacktraceObjectRegex.lastIndex = 0;
```

Comment:
let's just inline the regex to a local var here instead of having it be a prop. v8 optimizes this so it's just as fast as caching it.